### DISCUSSION TITLE
How to restrict "supabase db push"

### DISCUSSION THREAD

**Discussion by stefan-girlich** [supabase/supabase#21771] [Questions]
I want to set up local, staging, and prod environments as described in [this guide](https://supabase.com/docs/guides/cli/managing-environments?environment=ci). I want to make sure that a developer can pull changes from the remote staging or prod environment into their local setup, but not push migrations via `supabase db push` as seen in the linked article, as this seems like a way of bypassing Pull Requests in GitHub.

Is there a way to achieve this?

**Comment by valtteriluomapareto** [comment]
Would be also interested to hear if this is possible somehow.

**Comment by Magnuti** [comment]
Yeah this would be nice, especially now since Claude is very fond of running `supabase db push` instead of `supabase migration up --local`

**Comment by ezou-earthvisio** [comment]
Curious any solution to restrict supabase db push? I've create skills in Claude, but sometimes it still do supabase db push

**Comment by JustGoscha** [comment]
Anyone?
